Cookbook Recipe - Onboarding (#857)
uyuni-project/uyuni-docs: modules/cookbook/pages/recipe-onboarding.adoc

* Set a new hostname.
  Your new hostname should have a subdomain name and thus include at least two periods.
  In this example, we are using ``client1.suma.example``
+
----
hostnamectl set-hostname client1.suma.example
----
* Check that your change was successful:
+
----
hostnamectl
----
* Open {yast} and navigate to menu:Network Services[Hostnames].
Edit the hostname to match the one you just set, and click btn:[OK].
* In {yast}, navigate to menu:System[Network Settings] and go to the [guimenu]``Hostname/DNS`` tab.
In the [guimenu]``Static hostname`` field, type your new hostmame.
* Check that the change was successful:
+
----
hostname -f
----
+
This command should return your new FQDN.

Prepare software channels on the {productname} Server::

* In the {productname} {webui}, navigate to menu:Admin[Setup Wizard].
* In the [guimenu]``Organization Credentials`` tab, ensure you have entered your {scc} credentials, and are correctly authenticated.
* In the [guimenu]``Products`` tab, ensure that the product catalog is fully updated:
+
image::scc_sync.png[scaledwidth=80%]
* Use the product search bar to find the channels you need for your client operating system.

> Check the channels you want to install, and click btn:[Add products]:

Author: ```suggestion
  Check the channels you want to install, and click btn:[Add products]:
```

+
image::add_products.png[scaledwidth=80%]
* Wait for the product channels to fully synchronize.
  Depending on the products you have chosen, this could take a long time.

Create an activation key::

* In the {productname} {webui}, navigate to menu:Systems[Activation Keys], and click btn:[Create Key].
* Give your activation key a name, and select the base channel that matches the client you want to onboard.
This should be the product you just enabled:
+
image::activationkey_base.png[scaledwidth=80%]
* Check the child channels to include, and any add-on system types you want clients registered with this key to have.

>   If you want this activation key to be used for all clients by default, check [guimenu]``Universal Default``.

Author: ```suggestion
```

[[cookbook-recipe-onboarding]]
= Recipe: Onboarding

{productname} is all about managing client systems.
So one of the first things you need to do is onboard some clients.
This recipe shows you how to set up your {productname} Server to manage a new client, set up the software channels you need, and bootstrap the client using an activation key.

== Problem

This recipe shows you how to onboard a client to your {productname} Server.

The client can be running any Linux operating system.
For a list of supported client systems, see xref:installation:client-requirements.adoc[].

This is one of the first tasks you need to do when you set up {productname} for the first time, and you will probably have to do it many more times as you use the product.

> This recipe shows you how to onboard a single client, for a recipe about onboarding multiple clients at the same time, see xref:FIXME[].

Reviewer: ```suggestion
```
This xref is distracting.  If needed we can add it at the end of this chapter, imo.

== Solution

When you have completed this recipe, your client is onboarded, and it can be seen in the systems list of the {productname} {webui}.
You can then use {productname} to manage the client.

== Before you begin

Before you start, you should already have:

* A {productname} Server installed, which you can access using the {webui}.
* A client machine with an operating system installed, which you can access across the network that your {productname} Server is on, using SSH.
* Appropriate subscriptions from http://scc.suse.com for the products you are using.

This recipe uses a {sles} 15 SP2 operating system.
You can use other Linux operating systems, but some of the steps might be different.
For more information on onboarding other clients, see xref:client-configuration:registration-methods.adoc[].

== Steps

> Configure a fully-qualified domain name (FQDN) on your client::

Reviewer: The trailing two colons seem to be intentional.

Author: yep, to turn it into a deflist.

* On the client, at the command prompt, show the current hostname:
+
----
hostname -f
----
+
This command will probably return an error, or show something like ``localhost``.
* Set a new hostname.

>   Your new hostname should include two periods.

Reviewer: ```suggestion
  Your new hostname should have a subdomain name and thus include at least two periods.
```

  In this example, we are using ``client1.suma.example``
+
----
hostnamectl set-hostname client1.suma.example
----
* Check that your change was successful:
+
----
hostnamectl
----
* Open {yast} and navigate to menu:Network Services[Hostnames].
Edit the hostname to match the one you just set, and click btn:[OK].
* In {yast}, navigate to menu:System[Network Settings] and go to the [guimenu]``Hostname/DNS`` tab.
In the [guimenu]``Static hostname`` field, type your new hostmame.
* Check that the change was successful:
+
----
hostname -f
----
+
This command should return your new FQDN.

Prepare software channels on the {productname} Server::

* In the {productname} {webui}, navigate to menu:Admin[Setup Wizard].
* In the [guimenu]``Organization Credentials`` tab, ensure you have entered your {scc} credentials, and are correctly authenticated.

> * In the [guimenu]``Products`` tab, ensure that the product catalog is fully updated:

Reviewer: I wouldn't introduce the new term "product catalog" (or is the that used in the UI?).

Author: I think that's from the web UI, but I can double check.

Author: ![image](https://user-images.githubusercontent.com/3914967/113952754-a6a0cf00-9859-11eb-963a-948b849bfad7.png)

+
image::scc_sync.png[scaledwidth=80%]

> * Use the product search bar to find the channels you need for your client operating system.

Reviewer: Is this actually a separate step?

Author: This step includes entering a search term, checking the boxes, and clicking 'add products'. So I think that's a logical grouping.

> Check the channels you want to install, and click btn:[Add products]:

Reviewer: here, indentation probably is missing.

+
image::add_products.png[scaledwidth=80%]

> * Wait for the products to fully synchronize.

Reviewer: ```suggestion
* Wait for the product channels to fully synchronize.
```

  Depending on the products you have chosen, this could take a long time.

Create an activation key::

* In the {productname} {webui}, navigate to menu:Systems[Activation Keys], and click btn:[Create Key].
* Give your activation key a name, and select the base channel that matches the client you want to onboard.

> This should be the product you enabled in the last step:

Reviewer: "procedure" instead of "step"?

Author: What about this:

```suggestion
This should be the product you just enabled:
```

Or, maybe take it out entirely?

+
image::activationkey_base.png[scaledwidth=80%]
* Check the child channels to include, and any add-on system types you want clients registered with this key to have.

>   If you want this activation key to be used for all clients by default, check [guimenu]``Universal Default``.

Reviewer: I think we'd better warn about this feature.

Author: what do you mean?

Reviewer: AFAIK, it is not recommended to make use of this feature.  Thus, here in the context of the cookbook, we'd better ask the user to keep it unchecked.

Author: OK, I'll just take the sentence out, in that case.
